File: Server/three-d-portfolio-server/src/main/java/org/threeDPortfolioGallery/workloads/Exhibit.java
package org.threeDPortfolioGallery.workloads;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;

@Entity
public class Exhibit extends PanacheEntity {

    public String url;

    public String data_type;

    public String title;

    public String description;

    // relationships

    @JsonIgnore
    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    public Exhibition exhibition;

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    public Theme theme;

    @OneToOne
    public Position position;
}

File: Gallery/src/app/shared/class/exhibit.ts
import {Position} from "./position";

export class Exhibit{
  id : number;
  url : string;
  data_type : string;
  title : string;
  description : string;
  alignment: string | undefined
  position: Position | undefined
  scale: number | undefined

  constructor(id: number, model_url: string, data_type: string, title: string, desc: string, alignment: string | undefined, position: Position | undefined, scale: number | undefined) {
    this.id = id;
    this.url = model_url;
    this.data_type = data_type;
    this.title = title;
    this.description = desc;
    this.alignment = alignment
    this.position = position
    this.scale = scale
  }




}
